plutus-benchmark/bls12-381-costs/test/9.6/bls12-381-costs.golden

```diff
@@ -3,143 +3,143 @@ Hash n bytestrings onto G1 and add points
 
 
 Apply pairing to two pairs of points in G1 x G2 and run finalVerify on the results
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -     342   (2.1%)      1293505872  (12.9%)            4654   (0.0%) 
+    -     342   (2.1%)      1292150444  (12.9%)            4654   (0.0%) 
 
 
 Groth16 verification example
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -     779   (4.8%)      2316309514  (23.2%)           10904   (0.1%) 
+    -     779   (4.8%)      2312779174  (23.1%)           10904   (0.1%) 
 
 VRF example
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -     753   (4.6%)      1364980692  (13.6%)          184634   (1.3%) 
+    -     714   (4.4%)      1303368563  (13.0%)           49449   (0.4%) 
 
 G1 Verify
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -     332   (2.0%)      1465218237  (14.7%)            5754   (0.0%) 
+    -     332   (2.0%)      1463720946  (14.6%)            5754   (0.0%) 
 
 G2 Verify
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -     380   (2.3%)      1343563718  (13.4%)            5754   (0.0%) 
+    -     380   (2.3%)      1342066427  (13.4%)            5754   (0.0%) 
 
 Aggregate Single Key
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -     777   (4.7%)      3343214820  (33.4%)           71202   (0.5%) 
+    -     777   (4.7%)      3336910422  (33.4%)           71202   (0.5%) 
 
 Aggregate Multi Key
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -    1743  (10.6%)      3741726913  (37.4%)          566293   (4.0%) 
+    -    1701  (10.4%)      3676960887  (36.8%)          430886   (3.1%) 
 
 Schnorr Signature G1
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -     418   (2.6%)       361167811   (3.6%)          264488   (1.9%) 
+    -     372   (2.3%)       248182411   (2.5%)           13996   (0.1%) 
 
 Schnorr Signature G2
 
     n     Script size             CPU usage               Memory usage
   ----------------------------------------------------------------------
-    -     562   (3.4%)       606243489   (6.1%)          264656   (1.9%) 
+    -     516   (3.1%)       493258089   (4.9%)           14164   (0.1%) 
```

Comment:
1.9% to 0.1%? Does something get optimized away? Or does it now fail? Or why such a drastic difference?

Reply:
> 1.9% to 0.1%? Does something get optimized away? Or does it now fail? Or why such a drastic difference?

Good question.  You had me worried, but it's because the computation's chaining together a bunch of bls12-381 builtins and the memory costs of those are pretty small:
```
bls12_381_G1_add                : 18
bls12_381_G1_compress           : 6
bls12_381_G1_equal              : 1
bls12_381_G1_hashToGroup        : 18
bls12_381_G1_neg                : 18
bls12_381_G1_scalarMul          : 18
bls12_381_G1_uncompress         : 18
bls12_381_G2_add                : 36
bls12_381_G2_compress           : 12
bls12_381_G2_equal              : 1
bls12_381_G2_hashToGroup        : 36
bls12_381_G2_neg                : 36
bls12_381_G2_scalarMul          : 36
bls12_381_G2_uncompress         : 36
bls12_381_finalVerify           : 1
bls12_381_millerLoop            : 72
bls12_381_mulMlResult           : 72
```
The CPU costs are a lot bigger, so the proportional change in the total CPU cost is much smaller.

Reply:
I mean, this is what we'd expect since the DIY version is awful?

Reply:
Great to know, thanks!